### PR TITLE
Fix/message deletion issues

### PR DESCRIPTION
**Pull request by ThomasSession**
[SES-2817](https://optf.atlassian.net/browse/SES-2817) - Making sure the deletion is available in the app bar menu in all cases as the deletion logic is now handled in the VM
[SES-2811](https://optf.atlassian.net/browse/SES-2811) - Making sure we do not attempt to send messages marked as deleted
[SES-2818](https://optf.atlassian.net/browse/SES-2818) - Updating the logic obtaining the 'lastMessage' of a thread, taking into account whether they were marked as deleted or not
[SES-2464](https://optf.atlassian.net/browse/SES-2464) - Changed the logic when finishing a conversation activity from a deleted thread. It now calls finish again but the home screen will refresh the search on resume if the search is visible

### CODE REVIEW

**Review by SessionHero01 — APPROVED** [review]
LGTM

File: app/src/main/java/org/thoughtcrime/securesms/database/MmsSmsDatabase.java [hunk 1]
```diff
@@ -96,6 +99,14 @@ public MmsSmsDatabase(Context context, SQLCipherOpenHelper databaseHelper) {
     }
   }
 
+  public @Nullable MessageRecord getNonDeletedMessageForTimestamp(long timestamp) {
```
Comment (AL-Session) [inline app/src/main/java/org/thoughtcrime/securesms/database/MmsSmsDatabase.java:102]: Isn't a non-deleted message just a message? Perhaps `getMessageForTimestamp`?

Ah, we now have messages with a 'deleted' type in the database.... that's a kinda odd way to go about things, but another way you could think of it is that the message is kinda in the "Recycle Bin" / "Trash" whatever you want to call it, so could potentially be recovered if required (but we never do and just clean that stuff out).

So okay, I guess...

**Review by AL-Session — APPROVED** [review]
All good.

File: app/src/main/java/org/thoughtcrime/securesms/database/MmsSmsDatabase.java (hunk 1 above)
Comment (ThomasSession) [reply app/src/main/java/org/thoughtcrime/securesms/database/MmsSmsDatabase.java:102]: Yeah a message can be completely gone or marked as deleted. It still exists in the db and shows up as a special case message with the bin icon and a label like 'This message was deleted for everyone'

**Pull request merged**